pkg/metrics/instances.go

```diff
@@ -162,12 +161,17 @@ func (i *instancesMetric) Setup(ctx context.Context) error {
 	}
 	i.observations = append(i.observations, obs)
 
-	instruments := make([]instrument.Asynchronous, 0, len(i.observations))
 	for _, o := range i.observations {
-		instruments = append(instruments, o.Gauge)
+		_, err = meter.RegisterCallback(
+			func(ctx context.Context, observer metric.Observer) error {
+				return i.callback(ctx, observer)
+			},
+			o.Gauge)
+		if err != nil {
+			return err
+		}
```

Comment:
Why do we need this custom callback here?
```suggestion
	instruments := make([]instrument.Asynchronous, 0, len(i.observations))
	for _, o := range i.observations {
		instruments = append(instruments, o.Gauge)
	}
	_, err = meter.RegisterCallback(i.callback, instruments...)
	return err
```

Reply:
The function signature changed and it was not building anymore.

Reply:
yes, seems the swapped the parameters. For me its still unclear why you register multiple callbacks?

See:
https://github.com/open-telemetry/opentelemetry-go/pull/3587/commits/908d6e486afd4148a180cc17d1e85387c7a746b9#diff-600ae769b73b7ada58d7a7cfaf867d56778a91c8e36db76c5362670cdfb16e3eR284

Reply:
Changed